File: verds_col_2022/process/src/import_clean_20240817.R
#
# Authors:     Sebas and Nicholas
# Maintainers: Sebas and Nicholas
# Year: 2024
# =========================================

pacman::p_load(readr, dplyr, tidyr, janitor, logger, assertr, purrr, arrow,
               digest, writexl, sf)

if(dir.exists("/Users/sebas/OneDrive")) {
  
  input_1 <- "C:/Users/sebas/OneDrive/Documents/repositorio_cepap/CEPAP/verds_col_2022/process/input/Veredas_de_Colombia/Veredas_de_Colombia.shp"
  
  input_2 <- "C:/Users/sebas/OneDrive/Documents/repositorio_cepap/CEPAP/verds_col_2022/process/output/shapefile.veredas.arrow"
  
  output_1 <- "/Users/sebas/OneDrive/Documents/CEPAP/censo_agropecuarios_2014/Total_nacional(csv)/code_censo_agr_2014/output/resultados_incononzo_castillo.xlsx"
} else {
  input_1 <- "/Users/ruta_nico"
  
  output_2 <- "/Users/ruta_nico"
}

log_info("load data")
ir <- read_sf(input_1) %>% 
  clean_names()

File: mapas/verds_col_2022/process/src/import_clean_2024.08.17.jl
#
# Authors:     Sebas and Nicholas
# Maintainers: Sebas and Nicholas
# Year: 2024
# =========================================
# CEPAP/team_data/cifras_agro/igac_gini/input/GINI_MUNICIPAL_PERCENTILES.csv

using Pkg, DataFrames, Random, Dates, Tables, TidierFiles, Shapefile, ArchGDAL

Random.seed!(19481210)

println("load data--", now())
input_1 = "C:/Users/sebas/OneDrive/Documents/repositorio_cepap/CEPAP/verds_col_2022/process/input/Veredas_de_Colombia/Veredas_de_Colombia.shp"
output_1 = "C:/Users/sebas/OneDrive/Documents/repositorio_cepap/CEPAP/verds_col_2022/process/output/shapefile_munis.arrow"

veredas = Shapefile.Table(input_1)

veredas_2 = ArchGDAL.read(input_1)

layer = ArchGDAL.getlayer(veredas_2, 0)

df = DataFrame(layer)

plot(df)

println("clean names--", now())
rename!(munis, lowercase.(names(munis)))

println("export--", now())
write_arrow(munis, output_1)

println("done import_clean.jl--", now())

Shapefile.Polygon(input_1)